src/transformers/generation_utils.py

```diff
@@ -1309,6 +1311,17 @@ def generate(
                 "Diverse beam search cannot be used in sampling mode. Make sure that `do_sample` is set to `False`."
             )
 
+        if typical_p is not None:
+            if typical_p < 0 or typical_p > 1:
+                raise ValueError("Value of typical_p must be between 0 and 1.")
+            elif do_sample is False:
+                raise ValueError(
+                    "Decoder argument `typical_p` must be used in sampling mode. "
+                    "Make sure that `do_sample` is set to `True`."
+                )
+            elif is_group_beam_gen_mode is True:
+                raise ValueError("Decoder argument `typical_p` is not supported with beam groups.")
```

Comment:
Well, since I asked to remove the two checks above, this one can be moved below :D (after [this line](https://github.com/huggingface/transformers/blob/2df602870b9a7403109b470322926b48e366e3d1/src/transformers/generation_utils.py#L1471))